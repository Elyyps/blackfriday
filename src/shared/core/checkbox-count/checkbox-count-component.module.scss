.counter {
  font-size: 15px;
  color: #777777;
  margin-bottom: 12px;
}

.filter-bar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: white;

  cursor: pointer;

  input {
    font-weight: 600;
  }
}
