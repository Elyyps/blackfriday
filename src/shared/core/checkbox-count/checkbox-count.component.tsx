import * as React from "react";

import styles from "./checkbox-count-component.module.scss";
import { CheckboxComponent } from "../checkbox";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { useRef } from "react";

export interface ICheckboxCountProps {
  item: FilterItem;
  onChecked: (id: string) => void;
}

const CheckboxCount = (props: ICheckboxCountProps) => {
  const checkboxRef = useRef<HTMLDivElement>(null);

  const onCheckedOutside = (event: any, id: string) => {
    // Neccesary to make sure the click event isn't used multiple times.
    if (checkboxRef.current && checkboxRef.current.contains(event.target)) {
      return;
    }

    if (props.onChecked && id) {
      props.onChecked(id);
    }
  };

  const onChecked = (id: string) => {
    if (props.onChecked && id) {
      props.onChecked(id);
    }
  };

  return (
    <div
      role="button"
      id="outsideDiv"
      onClick={e => onCheckedOutside(e, props.item.id)}
      className={styles["filter-bar"]}
    >
      <CheckboxComponent onClick={() => onChecked(props.item.id)} isChecked={props.item.isSelected} ref={checkboxRef}>
        {props.item.displayName}
      </CheckboxComponent>
      <span className={styles["counter"]}>{props.item.totalAmount}</span>
    </div>
  );
};

export { CheckboxCount };
