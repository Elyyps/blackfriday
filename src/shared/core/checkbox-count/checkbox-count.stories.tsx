import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { CheckboxCount } from "./checkbox-count.component";
/* tslint:disable */
storiesOf("Checkbox Count", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <CheckboxCount
      item={{
        displayName: "",
        id: "1",
        isSelected: false,
        totalAmount: 22
      }}
      onChecked={() => {}}
    />
  ));
