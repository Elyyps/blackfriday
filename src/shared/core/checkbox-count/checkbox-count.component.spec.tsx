import React from "react";
import { shallow } from "enzyme";

import { CheckboxCount } from "@app/core/checkbox-count/checkbox-count.component";

describe("[CheckboxCount]", () => {
  it("should render non selected without crashing", () => {
    const renderedComponent = shallow(
      <CheckboxCount
        item={{
          displayName: "",
          id: "1",
          isSelected: false,
          totalAmount: 22
        }}
        onChecked={jest.fn()}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it("should render selected without crashing", () => {
    const renderedComponent = shallow(
      <CheckboxCount
        item={{
          displayName: "",
          id: "1",
          isSelected: true,
          totalAmount: 22
        }}
        onChecked={jest.fn()}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });
});
