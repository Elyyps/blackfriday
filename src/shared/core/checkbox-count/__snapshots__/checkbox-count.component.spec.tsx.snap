// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CheckboxCount] should render non selected without crashing 1`] = `
<div
  className="filter-bar"
  onClick={[Function]}
  role="button"
>
  <ForwardRef
    isChecked={false}
    onClick={[Function]}
  />
  <span
    className="counter"
  >
    22
  </span>
</div>
`;

exports[`[CheckboxCount] should render selected without crashing 1`] = `
<div
  className="filter-bar"
  onClick={[Function]}
  role="button"
>
  <ForwardRef
    isChecked={true}
    onClick={[Function]}
  />
  <span
    className="counter"
  >
    22
  </span>
</div>
`;
